engine/api/src/main/java/io/deephaven/engine/table/Table.java

```diff
@@ -363,6 +363,20 @@ public interface Table extends
     @ConcurrentMethod
     Table slice(long firstPositionInclusive, long lastPositionExclusive);
 
+    /**
+     * Extracts a subset of a table by row percentages.
+     * <p>
+     * Returns a subset of table in the range [startPercentInclusive * sizeOfTable , endPercentExclusive * sizeOfTable).
+     * For example, for a table of size 10, slicePct(0.1, 0.7) will return a subset from the second row to the seventh
+     * row. The percentage arguments must be in [0,1], otherwise the function returns an error.
+     *
+     * @param startPercentInclusive the starting percentage point for rows to include in the result, range [0, 1]
+     * @param endPercentExclusive the ending percentage point for rows to include in the result, range [0, 1]
+     * @return a new Table, which is the requested subset of rows from the original table
+     */
+    @ConcurrentMethod
+    Table slicePct(double startPercentInclusive, double endPercentExclusive);
```

Comment:
Looking at this spec, there is a weird case to consider.  `slicePct(0,1)` will return a table with `N-1` rows.  How would a user get all of the rows?  Seems impossible.  

... Actually, from reading the code, I _think_ that `slicePct(0,1)` will return the entire table, as desired.  This would indicate that the docs need to be reworked to reflect what the operation is actually doing.